library/yum_versionlock.py

```diff
@@ -15,117 +15,265 @@
+    # For debug purpose you can include the listed packages in the debug
+    #state_pkg['listed'] = []
+    missing_pkg_version = None
+    for is_checked in List_to_check:
+        if is_checked != 'Installed Packages' and is_checked != 'Available Packages':
+            # Split to concatenate name & version
+            pkg_split = is_checked.split()
+            if missing_pkg_version:
+                pkg_name = missing_pkg_version
+                pkg_version = pkg_split[0]
+                missing_pkg_version = None
+            else:
+                pkg_name = pkg_split[0].split('.')[0]
+                try:
+                    pkg_version = pkg_split[1]
+                except IndexError:
+                    missing_pkg_version = pkg_split[0].split('.')[0]
+            if pkg_version:
+                pkg_vers = pkg_name + '-' + pkg_version
+                # For debug purpose you can include the listed packages in the debug
+                #state_pkg['listed'].append(pkg_vers)
+                if check_type == 'installed':
+                    regex_search = re.compile('^([0-9]+:)?%s-([0-9]+:)?%s\.\*'%(pkg_name,pkg_version))
+                else:
+                    regex_search = re.compile('^\!([0-9]+:)?%s-([0-9]+:)?%s\.\*'%(pkg_name,pkg_version))
+                    # Init flag is_present
+                is_present = False
+                # search for the installed package in versionlock list
+                for locked in versionlock_packages:
+                    if regex_search.search(locked):
+                        is_present = True
+                if is_present:
+                    state_pkg['present'].append(pkg_vers)
+                else:
+                    state_pkg['missing'].append(pkg_vers)
+    return state_pkg
 
 
 def main():
-    """ start main program to add/remove a package to yum versionlock"""
+    """ start main program to add/exclude/delete a package to versionlock"""
+    default_mgr = __file__.split('_')[1]
     module = AnsibleModule(
         argument_spec=dict(
-            state=dict(required=True, type='str'),
-            package=dict(required=True, type='str'),
+            state=dict(required=True, type='str', choices=['present', 'excluded', 'absent', 'clear']),
+            package=dict(required=False, type='str'),
+            repo_mgr=dict(required=False, type='str', choices=['dnf', 'yum'], default=default_mgr),
         ),
-        supports_check_mode=False
+        supports_check_mode=True,
+        required_one_of=[['state']],
     )
 
     state = module.params['state']
     package = module.params['package']
-    changed = False
+    repo_mgr = module.params['repo_mgr']
 
-    # Check for yum version lock plugin
-    versionlock_plugin = get_state_yum_versionlock()
+    result = dict(
+        changed=False,
+        debug=dict(),
+        status=dict(),
+        diff=dict(),
+    )
+
+    # Check for the versionlock config file and if plugins is enabled
```

Comment:
Module will fail if there isn't any of them present:
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: IOError: [Errno 2] Datei oder Verzeichnis nicht gefunden: '/etc/yum/pluginconf.d/versionlock.conf'

Reply:
I will add a "try" to get a better behaviour when the file is absent.